### ISSUE TITLE
approle_role_create is not idempotent

### ISSUE THREAD

**Issue by drewmullen** [TerryHowe/ansible-modules-hashivault#136]
```yaml
- name: approle
  hashivault_approle_role_create:
    name: template
    bind_secret_id: true
    bound_cidr_list: []
    secret_id_num_uses: 0
    secret_id_ttl: 0
    token_num_uses: 0
    token_ttl: "10m"
    token_max_ttl: "15m"
    period: ""
    enable_local_secret_ids: false
    policies: ["default", "secret_rw"]
```

run 1
```
TASK [namespace : approle] ********************************
changed: [localhost]
```

run 2
```
TASK [namespace : approle] ********************************
changed: [localhost]
```

hopefully i can investigate at some point but documenting for now

*bonus issue: doesnt support `mount_point`

would be nice if it took a arguments as a file too

**Comment by codeselim** [comment]
mount_point is what is missing as described also in my raised issue: https://github.com/TerryHowe/ansible-modules-hashivault/issues/139

**Comment by drewmullen** [comment]
thanks for pointing out that issue. yeah, imo this should be rewritten as a new module

hashivault_approle_role
- take `state` argument (present, absent)
- catch for configuration changes (idempotency)
- take `mount_point` optional arg
- accept arguments as a file

id like to do this but realistically i probably wont prioritize for a bit personally. im happy to help coach if you have interest!

**Comment by codeselim** [comment]
@drewmullen Thank you for your fast answer. I will tell my team that and see if there is anyone who can help in the rewrite

**Comment by drewmullen** [comment]
alright! always good to give back :D lmk if i can help advise

**Comment by TerryHowe** [comment]
https://github.com/TerryHowe/ansible-modules-hashivault/pull/148

**Comment by drewmullen** [comment]
closed by #148 and #149
